src/EnergyPlus/ReportSizingManager.cc

```diff
@@ -3231,60 +3231,64 @@ namespace ReportSizingManager {
+                                            break;
+                                        }
+                                } // end switch
 
-                            switch (PrimaryAirSystem(CurSysNum).retFanModelTypeEnum) {
-                            case DataAirSystems::structArrayLegacyFanModels: {
-                                FanCoolLoad += (1.0 - OutAirFrac) * FanDesHeatGain(PrimaryAirSystem(CurSysNum).RetFanNum, DesVolFlow);
-                                break;
-                            }
-                            case DataAirSystems::objectVectorOOFanSystemModel: {
-                                FanCoolLoad += (1.0 - OutAirFrac) *
-                                               HVACFan::fanObjs[PrimaryAirSystem(CurSysNum).retFanVecIndex]->getFanDesignHeatGain(DesVolFlow);
-                                break;
-                            }
-                            case DataAirSystems::fanModelTypeNotYetSet: {
-                                // do nothing
-                                break;
-                            }
-                            } // end switch
+                                switch ( PrimaryAirSystem(CurSysNum).retFanModelTypeEnum ) {
+                                    case DataAirSystems::structArrayLegacyFanModels: {
+                                            FanCoolLoad += ( 1.0 - OutAirFrac ) * FanDesHeatGain(PrimaryAirSystem(CurSysNum).RetFanNum, DesVolFlow);
+                                            break;
+                                        }
+                                    case DataAirSystems::objectVectorOOFanSystemModel: {
+                                            FanCoolLoad += ( 1.0 - OutAirFrac ) *
+                                                HVACFan::fanObjs[ PrimaryAirSystem(CurSysNum).retFanVecIndex ]->getFanDesignHeatGain(DesVolFlow);
+                                            break;
+                                        }
+                                    case DataAirSystems::fanModelTypeNotYetSet: {
+                                            // do nothing
+                                            break;
+                                        }
+                                } // end switch
 
-                            PrimaryAirSystem(CurSysNum).FanDesCoolLoad = FanCoolLoad;
+                                PrimaryAirSystem(CurSysNum).FanDesCoolLoad = FanCoolLoad;
+                            }
                             PeakCoilLoad = max(0.0, (rhoair * DesVolFlow * (CoilInEnth - CoilOutEnth)));
                             CpAir = PsyCpAirFnW(CoilInHumRat);
                             // adjust coil inlet/outlet temp with fan temperature rise
                             if (DataDesAccountForFanHeat) {
-                                PeakCoilLoad += FanCoolLoad;
+                                PeakCoilLoad = max(0.0, (rhoair * DesVolFlow * (CoilInEnth - CoilOutEnth) + FanCoolLoad));
```

Comment:
Revised this since the max(0.0, xx) could cause a problem where += FanCoolLoad would be greater than capacity + fan heat.